pkg/embeddedcluster/upgrade.go

```diff
@@ -312,8 +335,44 @@ func getLocalArtifactMirrorImage(
 	return imageName, nil
 }
 
+func getEmbeddedClusterMetadataFromReplicatedApp(ctx context.Context, in *embeddedclusterv1beta1.Installation) ([]byte, error) {
+	var metadataURL string
+	if in.Spec.Config.MetadataOverrideURL != "" {
+		metadataURL = in.Spec.Config.MetadataOverrideURL
+	} else {
+		metadataURL = fmt.Sprintf(
+			"%s/embedded-cluster-public-files/metadata/v%s.json",
+			in.Spec.MetricsBaseURL,
```

Comment:
should use `util.ReplicatedAppEndpoint(license)`

Reply:
updated